### PR TITLE
Fix avatar loading on servers that prevent unauthenticated avatar access

### LINKED ISSUE #603: Avatars and other images don't display
- Your Rocket.Chat.ReactNative app version: 1.8.0.5534

- Your Rocket.Chat server version: 0.74.0

- Device model (or emulator) you're running with: iPhone 8

- Steps to reproduce: Connect to a server with security enabled for avatars, avatars don't load.

### PR DESCRIPTION
**Pull request by davidlougheed**
@RocketChat/ReactNative

Closes #603

Include auth information when accessing avatars. This issue affects all the apps and it's **surprisingly** old and unfixed thusfar. See https://github.com/RocketChat/Rocket.Chat.Android/issues/1749

### CODE REVIEW

**Comment by davidlougheed** [comment]
@AllanWang right now user isnt passed as a prop, so more work would be needed, but it might be good.

**Comment by davidlougheed** [comment]
@AllanWang may be worth checking out fixing in the android app too: https://github.com/RocketChat/Rocket.Chat.Android/blob/e6d7bceacdebad4d853011edc0c8937f2ac40401/app/src/main/java/chat/rocket/android/util/extensions/String.kt

**Comment by AllanWang** [comment]
No activity has happened in the android build since 15 days ago, and I (and others) have PRs pending. If activity resumes we may consider it.

**Comment by diegolmello** [comment]
> Nice job!
> Can you fix lint?
> Thanks.

@davidlougheed

**Comment by davidlougheed** [comment]
@diegolmello working on it. looks like I will need to pass `user` all the way down to Avatar, which will take a while to implement/test. incidentally will address point made by @AllanWang

**Comment by diegolmello** [comment]
@davidlougheed Don't worry about passing `user` down the tree.
Your lint error is a jest error actually.
And I think is `@rocketchat/sdk` not being mock.

**Comment by davidlougheed** [comment]
@diegolmello it may be worth hooking up avatar to state anyway, since that seems to be how user token/ids are accessed elsewhere, no?

**Comment by davidlougheed** [comment]
@diegolmello ok, I re-did it in a less cowboy-code way, and the linting/testing seems happier.

**Comment by diegolmello** [comment]
@davidlougheed Ok, I'll take a look.
Thanks for your fix 👊

File: app/presentation/RoomItem.js [hunk 1]
```diff
@@ -144,7 +148,11 @@ export default class RoomItem extends React.Component {
 		userMentions: PropTypes.number,
 		id: PropTypes.string,
 		onPress: PropTypes.func,
-		username: PropTypes.string,
+		user: PropTypes.shape({
+			id: PropTypes.string,
```
Comment (AllanWang) [inline app/presentation/RoomItem.js]: If you removed `username`, shouldn't you also remove `id`?

File: app/views/ProfileView/index.js [hunk 2]
```diff
@@ -437,6 +437,7 @@ export default class ProfileView extends LoggedView {
 								avatar={avatar && avatar.url}
 								size={100}
 								baseUrl={baseUrl}
+								user={user}
```
Comment (diegolmello) [inline app/views/ProfileView/index.js]: You should add `token` to `user`

Comment (davidlougheed) [reply app/views/ProfileView/index.js]: Done in 9299af2

File: app/views/ProfileView/index.js [hunk 3]
```diff
@@ -417,7 +418,7 @@ export default class ProfileView extends LoggedView {
 
 	render() {
 		const {
-			name, username, email, newPassword, avatarUrl, customFields, avatar, saving, showPasswordAlert
+			name, username, email, newPassword, avatarUrl, customFields, avatar, saving, showPasswordAlert, user
```
Comment (diegolmello) [inline app/views/ProfileView/index.js]: You've declared `user` from `state` instead of `props`

**Comment by davidlougheed** [comment]
I can't sign in anymore – not sure if this has anything to do with avatar stuff, or maybe the new SDK stuff. It logs handleSelectServer and doesn't actually let me sign in, and the button for selecting a server is always spinning.

**Comment by diegolmello** [comment]
@davidlougheed Yeah, new SDK demands Yarn because of crazy dependency chain.
`rm -rf node_modules` and `yarn` should solve.

**Comment by diegolmello** [comment]
Merged!
Thanks @davidlougheed for the nice PR and @AllanWang for the help reviewing it.

**Comment by davidlougheed** [comment]
thanks!

**Pull request merged**